src/groovy/com/unifina/api/UpdateProductCommand.groovy

```diff
@@ -24,13 +27,16 @@ class UpdateProductCommand {
 	Product.Currency priceCurrency
 	Long minimumSubscriptionInSeconds
 
+	PermissionService permissionService
```

Comment:
Hmm, looks sort of weird to have `permissionService` listed among the content fields. Would it theoretically be possible for an attacker to try and inject something to this field with a payload like:

```
{
  "permissionService": //something
  ...
}
```

If the above is at all a valid concern (I'm not saying it is), the service can also be accessed statically in `updateProduct` via `Holders.getApplicationContext().getBean(StreamrClientService.class)`.

Reply:
There was some (old) discussion on stackoverflow. I'll pass it as a parameter.